src/main/java/com/tfgm/models/Tram.java

```diff
@@ -72,4 +75,28 @@ public String toJSONString() {
         + '\"'
         + '}';
   }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Tram tram = (Tram) o;
+
+        if (!Objects.equals(uuid, tram.uuid)) return false;
+        if (!Objects.equals(endOfLine, tram.endOfLine)) return false;
+        if (!Objects.equals(destination, tram.destination)) return false;
+        if (!Objects.equals(origin, tram.origin)) return false;
+        return Objects.equals(tramNetworkDTO, tram.tramNetworkDTO);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = uuid != null ? uuid.hashCode() : 0;
```

Comment:
Probably don't want to include the ID in equals.

Reply:
Why?